### PR TITLE
Integration Test for Index Partitioning

### PR DESCRIPTION
**Pull request by vivianliu0**
Added a new integration test in test_encode.py which partitions the CACM index into two halves, searches them individually and merges hits to determine the top hit.

### CODE REVIEW

File: integrations/dense/test_encode.py [hunk 1]
```diff
@@ -58,6 +58,57 @@ def test_dpr_encode_as_faiss(self):
         self.assertAlmostEqual(hit[0].vectors[0], -6.88267112e-01, places=4)
         self.assertEqual(searcher.num_docs, 3204)
 
+    def test_dpr_encode_as_faiss_search_with_partitions(self):
+        # Create two partitions of the CACM index, search them individually, and merge results to compute top hit
+        index_dir = f'{self.pyserini_root}/temp_index'
+        os.makedirs(os.path.join(index_dir, 'partition1'), exist_ok=True)
+        os.makedirs(os.path.join(index_dir, 'partition2'), exist_ok=True)
+        self.temp_folders.append(index_dir)
+        cmd1 = f"python -m pyserini.encode input   --corpus {self.corpus_path} \
```
Comment (lintool) [inline integrations/dense/test_encode.py:67]: single quote please...

note sure why the above f-string is in double quotes, maybe oversight from previous PR - can you fix that also?

File: integrations/dense/test_encode.py [hunk 2]
```diff
@@ -58,6 +58,57 @@ def test_dpr_encode_as_faiss(self):
...
+
+        searcher_partition1 = FaissSearcher(
+            index_dir + '/partition1',
+            'facebook/dpr-question_encoder-multiset-base'
+        )
+        searcher_partition2 = FaissSearcher(
+            index_dir + '/partition2',
+            'facebook/dpr-question_encoder-multiset-base'
+        )
+        q_emb, hit1 = searcher_partition1.search("What is the solution of separable closed queueing networks?", k=2, return_vector=True)
```
Comment (lintool) [inline integrations/dense/test_encode.py:103]: single quotes also.

File: integrations/dense/test_encode.py [hunk 3]
```diff
@@ -58,6 +58,57 @@ def test_dpr_encode_as_faiss(self):
...
+        with open(os.path.join(index_dir, 'partition1/docid'), 'w') as docid1, open(os.path.join(index_dir, 'partition2/docid'), 'w') as docid2:
+            with open(os.path.join(index_dir, 'docid'), 'r') as file:
+                for i in range(index.ntotal):
+                    line = next(file)
+                    if i < (index.ntotal // 2):
+                        docid1.write(line)
+                    else:
+                        docid2.write(line)
+
+        searcher_partition1 = FaissSearcher(
```
Comment (lintool) [inline integrations/dense/test_encode.py:95]: I'm okay with long-ish lines (e.g., up to 120 wide) - so move on a single line?

File: integrations/dense/test_encode.py [hunk 4]
```diff
@@ -58,6 +58,57 @@ def test_dpr_encode_as_faiss(self):
...
+            index_dir + '/partition1',
+            'facebook/dpr-question_encoder-multiset-base'
+        )
+        searcher_partition2 = FaissSearcher(
+            index_dir + '/partition2',
+            'facebook/dpr-question_encoder-multiset-base'
+        )
+        q_emb, hit1 = searcher_partition1.search("What is the solution of separable closed queueing networks?", k=2, return_vector=True)
+        q_emb, hit2 = searcher_partition2.search("What is the solution of separable closed queueing networks?", k=2, return_vector=True)
+        mergedHits = hit1 + hit2
```
Comment (lintool) [inline integrations/dense/test_encode.py:105]: variables should be snake case, `merged_hits`

**Review by lintool — CHANGES_REQUESTED** [review]
Minor nits.

**Comment by vivianliu0** [comment]
> Minor nits.

Thanks for the comments, updated PR

**Review by lintool — APPROVED** [review]

**Pull request merged**